app/events/rtjc.go

```diff
@@ -24,22 +26,27 @@ var pinned = map[string]string{
 // Rtjc is a listener for incoming rtjc commands. Publishes whatever it got from the socket
 // compatible with the legacy rtjc bot. Primarily use case is to push news events from news.radio-t.com
 type Rtjc struct {
-	Port          int
-	Submitter     submitter
-	OpenAISummary openAISummary
+	Port       int
+	Submitter  submitter
+	Summarizer summarizer
 
-	UrAPI    string
-	UrToken  string
-	URClient *http.Client
+	Swg             *syncs.SizedGroup
+	SubmitRateLimit rate.Limit
+	SubmitRateBurst int
 }
 
 // submitter defines interface to submit (usually asynchronously) to the chat
 type submitter interface {
 	Submit(ctx context.Context, text string, pin bool) error
+	SubmitHTML(ctx context.Context, text string, pin bool) error
+}
+
+type summarizer interface {
+	GetSummariesByMessage(remarkLink string) (messages []string, err error)
 }
 
-type openAISummary interface {
-	Summary(text string) (response string, err error)
+type ioReader interface {
```

Comment:
this interface exists to simplify testing and has no other purpose. But this is Reade and there standard ways to provide this mock/dup with StringBuffer or byte.Buffer. Same thing as the mock, just no strange interface an no mock